src/filesystem/haiku/SDL_sysfilesystem.cc

```diff
@@ -99,4 +99,54 @@ SDL_GetPrefPath(const char *org, const char *app)
     return retval;
 }
 
+char *
+SDL_GetPath(SDL_Folder folder)
+{
+    const char *home = NULL;
+    char *retval;
+
+    home = SDL_getenv("HOME");
+    if (!home)
+    {
+        SDL_SetError("No $HOME environment variable available");
+        return NULL;
+    }
+
+    switch (folder)
+    {
+        case SDL_FOLDER_HOME:
+            retval = SDL_strdup(home);
+
+            if (!retval)
+                SDL_OutOfMemory();
+
+            return retval;
+
+            /* TODO: Is Haiku's desktop folder always ~/Desktop/ ? */
+        case SDL_FOLDER_DESKTOP:
+            retval = (char *) SDL_malloc(SDL_strlen(home) + 10);
+
+            if (!retval)
+                SDL_OutOfMemory();
+
+            SDL_strlcpy(retval, home, SDL_strlen(home) + 10);
+            SDL_strlcat(retval, "/Desktop/", SDL_strlen(home) + 10);
+
+            return retval;
+
+        case SDL_FOLDER_DOCUMENTS:
+        case SDL_FOLDER_DOWNLOADS:
+        case SDL_FOLDER_MUSIC:
+        case SDL_FOLDER_PICTURES:
+        case SDL_FOLDER_PUBLICSHARE:
+        case SDL_FOLDER_SAVEDGAMES:
+        case SDL_FOLDER_SCREENSHOTS:
+        case SDL_FOLDER_TEMPLATES:
+        case SDL_FOLDER_VIDEOS:
+        default:
+            SDL_SetError("Screenshots folder unavailable on XDG");
```

Comment:
```suggestion
            SDL_SetError("Unsupported folder type");
```

Reply:
Ah, good catch!

Is it a good practice to specify more details, like "Unsupported folder type on Haiku"?

Reply:
Perhaps, you could also use a simple `SDL_Unsupported()`.
I think the output of `SDL_GetError` is only informative, and should not be acted upon by a program (using `strstr` for example).

Reply:
I know, I just worry about the applications that print errors to the console or to log files. When a user reports buggy behavior in a large program, and the developer in charge of fixing the bug doesn't necessarily know where the bug comes from, I thought the bug could be understood more quickly with "Only HOME and DESKTOP available on Haiku" than just "Unsupported folder type".

That's at least how I see it - I'm willing to change that if needed.

Reply:
My thinking is SDL_Unsupported is more like "this operation is not available, or unimplemented, for this platform" ... like, I would take that to mean "this API isn't usable here" and not "this particular type of folder isn't known to this OS but other types might be if you try again"

fwiw.

Reply:
Sorry for the delay, I didn't see the message. I've updated the places to use SDL_Unsupported() where appropriate.

Reply:
I agree with @icculus, if a function is not available on the platform it should return SDL_Unsupported(), and where it's supported but not every value is available, more detailed error messages are appropriate.

I think you can undo the last commit, your code was fine and ready for merging.

Reply:
Ah, my bad, I misunderstood. I've reverted the last commit.